Add in SatChip doc (#12)
ASFHyP3/terramind-docs: severe-weather/SatChipDataModule_README.md

# Working with SatChip Datasets

> SatChip datasets are non-geospatial [TorchGeo](https://github.com/torchgeo/torchgeo?tab=readme-ov-file) datasets specialized for satellite imagery.

Reviewer: ```suggestion
SatChip datasets are pre-chipped [TorchGeo](https://github.com/torchgeo/torchgeo?tab=readme-ov-file) datasets specialized for satellite imagery.
```

SatChip prepares data labels and satellite images into 264x264 image arrays that follow the TerraMind extension of the MajorTom specification. These files are saved them as `zarr.zip` files and can be loaded in Python with `satchip.utils.load_chip()`. [Here](generating_chip_data.md) is a tutorial on generating your own chip data. 

Below is an example of metadata associated with the Sentinel-1 RTC chip `swathID_638_swathDate_2020-06-04`: 
```
<xarray.Dataset> Size: 587MB
Dimensions:   (sample: 263, platform: 1, time: 4, band: 2, y: 264, x: 264)
Coordinates:
  * band      (band) <U2 16B 'VH' 'VV'
  * platform  (platform) object 8B 'S1RTC'
  * sample    (sample) object 2kB '494U_807L_0_0' ... '507U_805L_1_3'
  * time      (time) datetime64[ns] 32B 2020-06-06T00:46:37 ... 2020-06-13T00...
  * x         (x) int64 2kB 0 1 2 3 4 5 6 7 ... 256 257 258 259 260 261 262 263
  * y         (y) int64 2kB 0 1 2 3 4 5 6 7 ... 256 257 258 259 260 261 262 263
Data variables:
    data      (sample, platform, time, band, y, x) float32 587MB ...
Attributes:
    date_created:     2020-06-04T00:00:00
    satchip_version:  0.3.0
    bounds:           [-103.24111, 44.39625, -101.30495, 45.58643]
```

> Pre-staged data are available in Google Drive to run this [tutorial](#example-tutorial). These data includes a Labeled hail damage, S1RTC, and S2L2A data from June 4, 2020.  

Reviewer: ```suggestion
Pre-staged data are available in Google Drive to run this [tutorial](#example-tutorial). These data include a Labeled hail damage, S1RTC, and S2L2A data from June 4, 2020.  
```

The `SatChip` dataset class is specifically designed for working with prepared SatChip data. It inherits TerraTorch's [`NonGeoDataset`](https://torchgeo.readthedocs.io/en/stable/api/datasets.html#non-geospatial-datasets) base class. Both the SatChip Dataset and DataModule code is in a [`loader.py`](loader.py).

Using the `SatChipDataModule` function, you can initialize the dataset by providing Path object paths to the labeled, S2L2A and S1RTC data.
```python
import loader

datamodule = loader.SatChipDataModule(batch_size=2, label_path=label_path, s2_path=s2_path, rtc_path=rtc_path)
```
This `DataModule` object can then be passed to the Pytorch `Trainer` object for training. 
We recommend using the [`TerraMind`](https://huggingface.co/ibm-esa-geospatial/TerraMind-1.0-base) when working with `SatChip` Data. You can check out the suggested model parameters in [`train.py`](train.py).

### Setup
If working on NAS for the first time, check out [this documentation](NAS/initial_setup.md) outlining setting up your NAS environment for the first time. On NAS, the pre-created `terramind` conda environment can be activated with
```

> conda activate terramind

Reviewer: ```suggestion
source activate terramind
```
Weirdly it's different in NAS.
